package.json

```diff
@@ -96,7 +96,7 @@
     "ts-node": "^8.2.0",
     "tslint": "^5.17.0",
     "typedoc": "^0.22.15",
-    "typescript": "^4.1.3"
+    "typescript": "~4.3.5"
```

Comment:
Can you do the typescript upgrade as a separate PR?

Reply:
@brendandburns - before this change, if I was to regenerate the package-lock file, it would fetch the latest typescript version, which fails the build.
I see 3 options here:
1-Keep the version currently in the lock file (4.1.3), which will require me to modify the TS version to '~4.13' or '4.1.3' in this PR
2-Follow the implementation of '^4.1.3' and try to fetch the latest compatible version of TS without migrating code (this is what I did)
3-Follow the implementation of '^4.1.3' and try to fetch the latest compatible version of TS, while migrating the code as necessary 

I believe that option 1 will be the best way to go, is that what you want me to do?

Reply:
Let's do option 1. Thanks